modules/helfi_etusivu_entities/src/Plugin/Block/SurveyBlock.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\helfi_etusivu_entities\Plugin\Block;
+
+use Drupal\Component\Utility\Xss;
+use Drupal\Core\Block\Attribute\Block;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Utility\Error;
+use Drupal\helfi_etusivu_entities\Plugin\ExternalEntities\StorageClient\Surveys;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+
+/**
+ * Provides a 'Surveys' block.
+ */
+#[Block(
+  id: "surveys",
+  admin_label: new TranslatableMarkup("Surveys"),
+)]
+final class SurveyBlock extends EtusivuEntityBlockBase {
+
+  /**
+   * Entity types supported by survey and the corresponding field names.
+   */
+  private const ENTITY_TYPE_FIELDS = [
+    'node' => 'field_survey_content_pages',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    $currentEntity = $this->getCurrentPageEntity(array_keys(self::ENTITY_TYPE_FIELDS));
+
+    if ($survey = $this->getSurvey($currentEntity)) {
+      return $this
+        ->entityTypeManager
+        ->getViewBuilder('node')
+        ->view($survey, 'default');
+    }
+
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts(): array {
+    return Cache::mergeContexts(parent::getCacheContexts(), [
+      'url.path',
+      'languages:language_content',
+    ]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags(): array {
+    return Cache::mergeTags(parent::getCacheTags(), [
+      Surveys::$customCacheTag,
+      'node_list:survey',
+    ]);
+  }
+
+  /**
+   * Loads most resent survey node that is valid for this page.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface|null $currentEntity
+   *   Entity on current page.
+   *
+   * @returns NodeInterface|null
+   *   Most recent survey for this page or NULL if none exists.
+   */
+  public function getSurvey(?EntityInterface $currentEntity) : ?NodeInterface {
```

Comment:
This has some mentions of "announcements" in the comments which should be replaced with "surveys".